frontend/tests/features/wheresMyGene.test.ts

```diff
@@ -88,207 +82,132 @@ describe("Where's My Gene", () => {
     await expect(filtersPanel).toHaveSelector(getText("Sex"));
 
     // Legend
-    const Legend = await page.$("*css=div >> text=Gene Expression");
-
+    const Legend = page.getByTestId("legend-wrapper");
     await expect(Legend).toHaveSelector(getText("Gene Expression"));
     await expect(Legend).toHaveSelector(getText("Expressed in Cells (%)"));
 
     // Info Panel
-    const InfoPanel = await page.$("*css=div >> text=Filters");
-
-    await expect(InfoPanel).toHaveSelector(getText("Methodology"));
-    await expect(InfoPanel).toHaveSelector(
+    await expect(filtersPanel).toHaveSelector(getText("Methodology"));
+    await expect(filtersPanel).toHaveSelector(
       getText("After filtering cells with low coverage ")
     );
-    await expect(InfoPanel).toHaveSelector(getText("Source Data"));
+    await expect(filtersPanel).toHaveSelector(getText("Source Data"));
   });
 
   test("Filters and Heatmap", async ({ page }) => {
     await goToPage(`${TEST_URL}${ROUTES.WHERE_IS_MY_GENE}`, page);
 
-    async function getTissueSelectorButton() {
-      return page.$(getTestID(ADD_TISSUE_ID));
-    }
-
-    async function getGeneSelectorButton() {
-      return page.$(getTestID(ADD_GENE_ID));
-    }
-
-    await clickUntilOptionsShowUp(getTissueSelectorButton, page);
+    await clickUntilOptionsShowUp({ page, testId: ADD_TISSUE_ID });
     await selectFirstOption(page);
 
-    await clickUntilOptionsShowUp(getGeneSelectorButton, page);
+    await clickUntilOptionsShowUp({ page, testId: ADD_GENE_ID });
     await selectFirstOption(page);
 
     await waitForHeatmapToRender(page);
 
-    const sexSelector = await getSexSelector();
-
-    if (!sexSelector) throw Error("No sexSelector found");
-
-    const selectedSexesBefore = await sexSelector.$$(MUI_CHIP_ROOT);
+    const sexSelector = getSexSelector();
```

Comment:
we need to await getSexSelector(), since `page. getByTestId()` returns a promise

Reply:
It returns a locator, not a promise.
<img width="855" alt="image" src="https://user-images.githubusercontent.com/16548075/228383088-47c45e00-8446-45af-868c-99a3d5a1605f.png">

Reply:
Checked https://playwright.dev/docs/locators, Alec is right!